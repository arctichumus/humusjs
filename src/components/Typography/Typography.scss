.typography {
  text-align: center;
  font-style: normal;
  line-height: normal;
  font-family: 'Roboto', Arial, sans-serif;
  /* why this doesn't work? */
  font-weight: 400;

  &--upper-case {
    text-transform: uppercase;
  }

  &__Header {
    font-size: 5rem;
    // font-family: 'Roboto', Arial, sans-serif;
    font-weight: 400;
  }

  &__Subheader {
    font-size: 2.5rem;
    line-height: 2.5rem;
  }

  &__Text {
    font-size: 2rem;
    line-height: 2rem;
    letter-spacing: 0.1px;
  }

  &__Comment {
    font-size: 2rem;
    line-height: 2rem;
    letter-spacing: 0.1px;
  }
}