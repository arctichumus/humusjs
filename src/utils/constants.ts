export enum Color {
  /* Primary color */
  p0 = '#000000',
  p10 = '#21005D',
  p20 = '#381E72',
  p30 = '#4F378B',
  p40 = '#6750A4',
  p50 = '#7F67BE',
  p60 = '#9A82DB',
  p70 = '#B69DF8',
  p80 = '#D0BCFF',
  p90 = '#EADDFF',
  p100 = '#FFFFFF',
};

export enum TextAlign {
  center = 'center',
  left = 'left',
  right = 'right',
}

export enum TypographyVariant {
  Header = 'h1',
  Subheader = 'h2',
  Text = 'div',
  Comment = 'span',
}