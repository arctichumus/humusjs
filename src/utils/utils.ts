export function getKeyByValue<T extends Record<string, string>>(enumObj: T, value: string): keyof T | undefined {
  const keys = Object.keys(enumObj) as Array<keyof T>;

  for (const key of keys) {
    if (enumObj[key] === value) {
      return key;
    }
  }

  return undefined;
}
