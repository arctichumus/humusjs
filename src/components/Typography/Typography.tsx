import React, { CSSProperties, useMemo } from 'react';
import classnames from 'classnames/bind';
import 'typeface-roboto';

import { TypographyVariant, Color, TextAlign } from '../../utils/constants';
import { getKeyByValue } from '../../utils/utils';
import styles from './Typography.scss';

const cx = classnames.bind(styles);

export interface TextProps {
  color?: Color;
  variant?: TypographyVariant,
  textAlign?: TextAlign,
  upperCase?: boolean,
  style?: CSSProperties,
  className?: string,
  children?: React.ReactNode,
}

const Typography = ({
  color = Color.p0,
  variant = TypographyVariant.Subheader,
  textAlign = TextAlign.center,
  upperCase = false,
  style,
  className,
  children
}: TextProps) => {
  const WrapperTag = variant;
  const tagEnumName = useMemo(() => getKeyByValue(TypographyVariant, variant), [variant]);

  return (
    <WrapperTag
      className={cx(
        'typography',
        `typography__${tagEnumName}`,
        upperCase && 'typography--upper-case',
        className
      )}
      style={{
        color: color || Color.p0,
        textAlign: textAlign,
        ...style
      }}
    >
      {children}
    </WrapperTag>
  );
};

export default Typography;
