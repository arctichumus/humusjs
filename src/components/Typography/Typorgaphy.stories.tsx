import React from "react";
import { StoryFn, Meta } from "@storybook/react";

import { TypographyVariant, Color, TextAlign } from '../../utils/constants';

import Typography from "./Typography";

// More on default export: https://storybook.js.org/docs/react/writing-stories/introduction#default-export
export default {
  title: "HumusJS/Typography",
  component: Typography,
} as Meta<typeof Typography>;

const Template: StoryFn<typeof Typography> = (args) => <Typography {...args} />;

export const Lorem = Template.bind({});
Lorem.args = {
  variant: TypographyVariant.Text,
  color: Color.p0,
  textAlign: TextAlign.center,
  upperCase: false,
  children: 'Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua.',
  style: {},
  className: '',
};