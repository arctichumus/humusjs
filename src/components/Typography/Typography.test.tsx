import React from 'react';
import { render } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';

import Typography from './Typography';
import { TypographyVariant, Color, TextAlign } from '../../utils/constants';

describe('Typography', () => {
  it('renders children correctly', () => {
    const { getByText } = render(
      <Typography>
        <span>Test Text</span>
      </Typography>
    );

    const textElement = getByText('Test Text');
    expect(textElement).toBeInTheDocument();
  });

  it('applies correct className based on variant prop', () => {
    const { container } = render(<Typography variant={TypographyVariant.Header} />);
    expect(container.firstChild).toHaveClass('typography__Header');
  });

  it('applies correct color style based on color prop', () => {
    const { container } = render(<Typography color={Color.p10} />);
    expect(container.firstChild).toHaveStyle({ color: '#21005D' });
  });

  it('applies correct textAlign style based on textAlign prop', () => {
    const { container } = render(<Typography textAlign={TextAlign.left} />);
    expect(container.firstChild).toHaveStyle({ textAlign: 'left' });
  });

  it('applies correct case style based on upperCase prop', () => {
    const { container } = render(<Typography upperCase={true} />);
    console.log(container.firstChild)
    expect(container.firstChild).toHaveClass('typography--upper-case');
  });

  it('matches snapshot', () => {
    const { container } = render(<Typography />);
    expect(container.firstChild).toMatchSnapshot();
  });
});
