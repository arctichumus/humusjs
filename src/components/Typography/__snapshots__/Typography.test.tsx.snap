// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Typography matches snapshot 1`] = `
<h2
  class="typography typography__Subheader"
  style="color: rgb(0, 0, 0); text-align: center;"
/>
`;
